### PR TITLE
Check in/50995/day of details

### PR DESCRIPTION
**Pull request by brianseek**
## Summary
Adds new VAOS style to day-of experience with details page.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#50995

## Testing done

Unit and e2e tests

## Acceptance criteria

- [ ] New VAOS style shows behind feature flag
- [ ] Details page exists and you can check-in from it
- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: src/applications/check-in/components/AppointmentBlockVaos.jsx [hunk 1]
```diff
@@ -23,25 +28,41 @@ const AppointmentBlockVaos = props => {
...
-            <AppointmentListItem
+            <AppointmentListItemVaos
               key={`${appointment.appointmentIen}-${appointment.stationNo}`}
               appointment={appointment}
-              goToDetails={page === 'confirmation' ? handleDetailClick : null}
+              showDetailsLink={page === 'confirmation' || page === 'details'}
+              goToDetails={handleDetailClick}
+              AppointmentAction={
+                <AppointmentActionVaos
+                  appointment={appointment}
```
Comment (brianseek) [inline src/applications/check-in/components/AppointmentBlockVaos.jsx:60]: Prettier deletes the parentheses ¯\_(ツ)_/¯

File: src/applications/check-in/components/AppointmentDisplay/AppointmentActionVaos.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  const setSelectedAppointment = useCallback(
+    appt => {
+      dispatch(appointmentWasCheckedInto(appt));
+    },
+    [dispatch],
+  );
+  const { updateError } = useUpdateError();
+
+  const { jumpToPage } = useFormRouting(router);
+  const onClick = useCallback(
```
Comment (wullaski) [inline src/applications/check-in/components/AppointmentDisplay/AppointmentActionVaos.jsx:31]: Could we move the onclick and the dispatch to the AppointmentBlockVaos and pass the onclick through as a prop? That would probably remove the need for this file. I don't mind having all the code in the parent component. That way we could avoid having to pass the router down or the component as a prop.

**Review by wullaski — CHANGES_REQUESTED** [review]

File: src/applications/check-in/components/AppointmentDisplay/AppointmentActionVaos.jsx (hunk 2 above)
Comment (brianseek) [reply src/applications/check-in/components/AppointmentDisplay/AppointmentActionVaos.jsx:31]: yeah that makes sense to me!

**Review by wullaski — APPROVED** [review]
LGTM!

**Pull request merged**